### PR TITLE
CAF-2660 Audit Webservice Client implmentation

### PR DESCRIPTION
**Pull request by CJComac**
Outstanding: HTTP and HTTPS proxy support

### CODE REVIEW

File: caf-audit-service-container/pom.xml [hunk 1]
```diff
@@ -339,7 +337,7 @@
                                 <env>
                                     <CAF_APPNAME>test/auditservice</CAF_APPNAME>
                                     <CAF_CONFIG_PATH>/config</CAF_CONFIG_PATH>
-                                    <CAF_ELASTIC_HOST_AND_PORT>${docker.host.address}:${elasticsearch.transport.port}</CAF_ELASTIC_HOST_AND_PORT>
+                                    <CAF_ELASTIC_HOST_AND_PORT>elasticsearch:9300</CAF_ELASTIC_HOST_AND_PORT>
```
Comment (glucy) [inline caf-audit-service-container/pom.xml:340]: Should this be hardcoded?

File: caf-audit-service-container/src/integration-test/java/com/hpe/caf/services/audit/api/AuditConnectionHelper.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import com.hpe.caf.api.ConfigurationSource;
+import com.hpe.caf.auditing.AuditConnection;
+import com.hpe.caf.auditing.AuditConnectionFactory;
+import com.hpe.caf.auditing.elastic.ElasticAuditConfiguration;
+import com.hpe.caf.auditing.webserviceclient.WebserviceClientAuditConfiguration;
+
+public class AuditConnectionHelper
+{
+
+    public static AuditConnection getElasticAuditConnection(String esHostAndPorts, String esClusterName)
```
Comment (glucy) [inline caf-audit-service-container/src/integration-test/java/com/hpe/caf/services/audit/api/AuditConnectionHelper.java:28]: JavaDoc, this is a public method, it must have JavaDoc

File: caf-audit-service-container/src/integration-test/java/com/hpe/caf/services/audit/api/AuditConnectionHelper.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            {
+                ElasticAuditConfiguration elasticAuditConfiguration = new ElasticAuditConfiguration();
+                elasticAuditConfiguration.setHostAndPortValues(esHostAndPorts);
+                elasticAuditConfiguration.setClusterName(esClusterName);
+                return (T) elasticAuditConfiguration;
+            }
+        });
+    }
+
+    public static AuditConnection getWebserviceAuditConnection(String wsHostAndPort)
```
Comment (glucy) [inline caf-audit-service-container/src/integration-test/java/com/hpe/caf/services/audit/api/AuditConnectionHelper.java:46]: JavaDoc, this is a public method, it must have JavaDoc

File: caf-audit/src/main/java/com/hpe/caf/auditing/AuditConnectionFactory.java [hunk 4]
```diff
@@ -18,17 +18,37 @@
...
+
+        // If the AUDIT_LIB_MODE environment variable has not been set return the NOOP implementation
+        if (auditLibMode == null) {
+            return new NoopAuditConnection(configSource);
+        }
+
+        // Return WebServiceClient or Direct to Elastic search impl depending on AUDIT_LIB_MODE's value
+        if (auditLibMode.equals("webserviceclient")) {
+            return new WebserviceClientAuditConnection(configSource);
+        } else if (auditLibMode.equals("elasticsearchdirect")) {
```
Comment (glucy) [inline caf-audit/src/main/java/com/hpe/caf/auditing/AuditConnectionFactory.java]: Use .equalsIgnoreCase()

"else" is redundant, remove it

File: caf-audit-service-container/src/integration-test/java/com/hpe/caf/services/audit/api/WebserviceClientAuditIT.java [hunk 5]
```diff
@@ -0,0 +1,414 @@
...
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.*;
+import java.util.concurrent.ExecutionException;
+
+public class WebserviceClientAuditIT {
+
+    private static final String APPLICATION_ID = "aTestApplication";
+    private static final String TENANT_ID = "tTestTenant";
+    private static String USER_ID = "aTestUser@testcompany.com";
```
Comment (glucy) [inline caf-audit-service-container/src/integration-test/java/com/hpe/caf/services/audit/api/WebserviceClientAuditIT.java]: final

File: caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+    }
+
+    @Override
+    public void send() throws Exception {
+
+        //  Get the constructed Audit Event as a JSON string
+        String auditEventJson = getAuditEventAsJsonString();
+
+        // Reset the JsonBuilder
+        jsonBuilder = XContentFactory.jsonBuilder();
```
Comment (glucy) [inline caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java]: Should this be in the finally block?
Why is this being "reset" here?

File: caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java [hunk 7]
```diff
@@ -0,0 +1,273 @@
...
+        if (auditEventJson == null || auditEventJson.isEmpty()) {
+            String errorMessage = "No Audit Event JSON to send to the Audit webservice";
+            LOG.error(errorMessage);
+            throw new WebserviceClientException(errorMessage);
+        }
+
+        URL url = new URL("http://" + webserviceHostAndPort + "/caf-audit-service/v1/auditevents");
+        HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+        //  Try to send Audit Event to the Audit Webservice HTTP endpoint
+        try {
```
Comment (glucy) [inline caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java:159]: try has no catch?

catch specific exceptions, log them and re-throw if necessary, currently everything is exiting as java.lang.Exception with nothing being caught or logged

File: caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java [hunk 8]
```diff
@@ -0,0 +1,273 @@
...
+            conn.setConnectTimeout(5000);
+            conn.setRequestProperty("Content-Type", "application/json; charset=UTF-8");
+            conn.setDoOutput(true);
+
+            byte[] auditEventJsonBytes = auditEventJson.getBytes("UTF-8");
+
+            //  The number of bytes to send is known; set fixed length streaming mode
+            conn.setFixedLengthStreamingMode(auditEventJsonBytes.length);
+
+            OutputStream outputStream = conn.getOutputStream();
```
Comment (glucy) [inline caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java:169]: Could use try-with-resources for this.
Minimum try/catch.

File: caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java [hunk 9]
```diff
@@ -0,0 +1,273 @@
...
+            outputStream.write(auditEventJsonBytes);
+            outputStream.close();
+
+            int responseCode = conn.getResponseCode();
+
+            // Check that the response code was returned as expected. Print a WARN if the response code was a 200 but
+            // not 204. Print an error if the response code is anything else.
+            int expectedResponseCode = 204;
+            if (responseCode >= 200 && responseCode < 300) {
```
Comment (glucy) [inline caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java:178]: It is our web service.  If is does not return a 204 we have something unexpected.  No need to do this range.

File: caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java [hunk 10]
```diff
@@ -0,0 +1,273 @@
...
+                            + " when the expected response code is " + expectedResponseCode);
+                }
+            } else {
+                String errorMessage = "Webservice returned response code " + responseCode + " when the expected " +
+                        "response code is " + expectedResponseCode;
+                LOG.error(errorMessage);
+                throw new WebserviceClientException(errorMessage);
+            }
+            LOG.info("Audit event request sent and received response code " + responseCode + " from the webservice");
+        } finally {
```
Comment (glucy) [inline caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java:190]: As mentioned above, this method needs try-with-resources or try/catch blocks at a minimum.  When re-worked this may affect the content of the finally block.

File: pom.xml [hunk 11]
```diff
@@ -109,7 +109,7 @@
     </dependencyManagement>
 
     <modules>
-        <module>docs</module>
+        <!--<module>docs</module>-->
```
Comment (glucy) [inline pom.xml:112]: Undo before check in

File: caf-audit/src/main/java/com/hpe/caf/auditing/AuditConnectionFactory.java [hunk 12]
```diff
@@ -18,17 +18,37 @@
...
-        return new ElasticAuditConnection(configSource);
+        String auditLibMode = System.getProperty("AUDIT_LIB_MODE", System.getenv("AUDIT_LIB_MODE"));
+
+        // If the AUDIT_LIB_MODE environment variable has not been set return the NOOP implementation
+        if (auditLibMode == null) {
+            return new NoopAuditConnection(configSource);
+        }
+
+        // Return WebServiceClient or Direct to Elastic search impl depending on AUDIT_LIB_MODE's value
+        if (auditLibMode.equals("webserviceclient")) {
```
Comment (glucy) [inline caf-audit/src/main/java/com/hpe/caf/auditing/AuditConnectionFactory.java]: Use .equalsIgnoreCase()

lets keep this short so either:
* webservice
* direct

File: caf-audit-service-container/pom.xml [hunk 13]
```diff
@@ -275,6 +270,9 @@
                                     <Git.Branch>${git.branch}</Git.Branch>
                                     <Git.Commit>${git.revision}</Git.Commit>
                                 </labels>
+                                <env>
+                                    <AUDIT_LIB_MODE>elasticsearchdirect</AUDIT_LIB_MODE>
```
Comment (glucy) [inline caf-audit-service-container/pom.xml:274]: Lets keep this short
* webservice
OR
* direct

File: caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java [hunk 14]
```diff
@@ -0,0 +1,273 @@
...
+
+    private final String webserviceHostAndPort;
+
+    private final Map<String, Object> auditEventCommonFields = new HashMap<>();
+
+    private final List<EventParam> auditEventParams = new ArrayList<>();
+
+    private XContentBuilder jsonBuilder;
+
+    public WebserviceClientAuditEventBuilder(final String webserviceHostAndPort,
```
Comment (glucy) [inline caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java]: JavaDoc

File: caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java [hunk 15]
```diff
@@ -0,0 +1,273 @@
...
+        auditEventParams.add(new EventParam(name, "boolean", columnName, value));
+    }
+
+    @Override
+    public void addEventParameter(String name, String columnName, Date value) {
+        auditEventParams.add(new EventParam(name, "date", columnName, value));
+    }
+
+    @Override
+    public void send() throws Exception {
```
Comment (glucy) [inline caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java]: JavaDoc

**Review by glucy — CHANGES_REQUESTED** [review]

File: caf-audit-service-container/src/integration-test/java/com/hpe/caf/services/audit/api/AuditConnectionHelper.java (hunk 2 above)
Comment (mulhollandc) [reply caf-audit-service-container/src/integration-test/java/com/hpe/caf/services/audit/api/AuditConnectionHelper.java:28]: Minor point here but to me web service is two words so I would capitalise S in service for method names etc.

File: caf-audit/src/main/java/com/hpe/caf/auditing/AuditConnectionFactory.java (hunk 4 above)
Comment (mulhollandc) [reply caf-audit/src/main/java/com/hpe/caf/auditing/AuditConnectionFactory.java]: or could replace with switch and throw exception as default option.

File: caf-audit/src/main/java/com/hpe/caf/auditing/AuditConnectionFactory.java (hunk 12 above)
Comment (mulhollandc) [reply caf-audit/src/main/java/com/hpe/caf/auditing/AuditConnectionFactory.java]: Yes I agree a refactoring from WebServiceClient to just WebService

File: caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java [hunk 16]
```diff
@@ -0,0 +1,273 @@
...
+
+    private static final Logger LOG = LogManager.getLogger(WebserviceClientAuditEventBuilder.class.getName());
+
+    private final String webserviceHostAndPort;
+
+    private final Map<String, Object> auditEventCommonFields = new HashMap<>();
+
+    private final List<EventParam> auditEventParams = new ArrayList<>();
+
+    private XContentBuilder jsonBuilder;
```
Comment (mulhollandc) [inline caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java]: Should this not be local to getAuditEventAsJsonString() only?

File: caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java [hunk 17]
```diff
@@ -0,0 +1,273 @@
...
+
+    public WebserviceClientAuditEventBuilder(final String webserviceHostAndPort,
+                                             final AuditCoreMetadataProvider coreMetadataProvider) {
+        this.webserviceHostAndPort = webserviceHostAndPort;
+
+        try {
```
Comment (mulhollandc) [inline caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java]: I'm not sure why this is being done in the constructor.

File: caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java [hunk 18]
```diff
@@ -0,0 +1,273 @@
...
+            this.jsonBuilder = XContentFactory.jsonBuilder();
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (mulhollandc) [inline caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java]: If there is any chance of an exception, should we not be at least logging it?

File: caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java [hunk 19]
```diff
@@ -0,0 +1,273 @@
...
+        jsonBuilder = XContentFactory.jsonBuilder();
+
+        //  An Audit Event in Json format was not returned, throw exception
+        if (auditEventJson == null || auditEventJson.isEmpty()) {
+            String errorMessage = "No Audit Event JSON to send to the Audit webservice";
+            LOG.error(errorMessage);
+            throw new WebserviceClientException(errorMessage);
+        }
+
+        URL url = new URL("http://" + webserviceHostAndPort + "/caf-audit-service/v1/auditevents");
```
Comment (mulhollandc) [inline caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java:156]: This line will need changed in the future if we ever modify audit web service contract version.

File: caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java (hunk 9 above)
Comment (mulhollandc) [reply caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java:178]: The web service is only expect to return a 204 for success. Anything else would indicate something has gone wrong. We should log and throw an exception.

File: caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java [hunk 20]
```diff
@@ -0,0 +1,273 @@
...
+        }
+        jsonBuilder.endObject();
+        jsonBuilder.close();
+
+        return jsonBuilder.string();
+    }
+
+    private class EventParam {
+
+        public EventParam(String paramName, String paramType, String paramColumnName,
```
Comment (mulhollandc) [inline caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java:223]: Don't think we are done here. See schema change in CAF-2693.

File: caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientException.java [hunk 21]
```diff
@@ -0,0 +1,22 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hpe.caf.auditing.webserviceclient;
+
+public class WebserviceClientException extends Exception {
```
Comment (mulhollandc) [inline caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientException.java:18]: Why is this needed?

File: caf-audit/src/test/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditChannelTest.java [hunk 22]
```diff
@@ -0,0 +1,71 @@
...
+
+    @Test
+    public void testCreateEventBuilderWithAuditCoreMetadataProvider() throws Exception {
+        AuditCoreMetadataProvider acmp = AuditNewEventFactory.createNewEvent();
+        WebserviceClientAuditChannel channel = new WebserviceClientAuditChannel(testHostAndPort);
+        AuditEventBuilder auditEventBuilder = channel.createEventBuilder(acmp);
+        Assert.assertNotNull(auditEventBuilder);
+    }
+
+    @Test(expected = UnknownHostException.class)
```
Comment (mulhollandc) [inline caf-audit/src/test/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditChannelTest.java:57]: Why is this test here? Should this class not be for unit tests only? You are testing a number of calls here. Further, are we missing an integration test class or have I overlooked something?

**Review by mulhollandc — CHANGES_REQUESTED** [review]

File: caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java (hunk 16 above)
Comment (glucy) [reply caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java]: Agree with Connor.  Looks like jsonBuilder can be local to the method getAuditEventAsJsonString().

File: caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java (hunk 17 above)
Comment (glucy) [reply caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java]: If jsonBuilder can be local to the method getAuditEventAsJsonString() it will need removed from the constructor.  I can see no reason for doing this in the constructor at this time.

File: caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java (hunk 18 above)
Comment (glucy) [reply caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java]: This code is going to be removed but we should print to console, always use logs.

File: caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java (hunk 6 above)
Comment (glucy) [reply caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java]: This is probably going to move into the method getAuditEventAsJsonString(), correct?

File: caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java (hunk 20 above)
Comment (glucy) [reply caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java:223]: @CJComac and @mulhollandc you both may need to work together to sort out this little wrinkle.   Shouldn't be too difficult but Chris' changes will depend on code Connor is now putting through review and QA.

File: caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientException.java (hunk 21 above)
Comment (glucy) [reply caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientException.java:18]: I like this, but then again I'm like that :-)

File: caf-audit/src/test/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditChannelTest.java (hunk 22 above)
Comment (glucy) [reply caf-audit/src/test/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditChannelTest.java:57]: If this test is doing what I think it is doing this might be OK here as a Unit Test.

Basically my interruption is; sending a message to an unknown host which will cause an exception to be thrown immediately?  However I don't see immediately where the UnknownHostException comes from in Chris' code.  As at the moment the send() method just throws Exception.

File: caf-audit-service-container/pom.xml (hunk 1 above)
Comment (CJComac) [reply caf-audit-service-container/pom.xml:340]: Yes I have configured private networking between the webservice and elasticsearch instances with the `<links>` element.

File: caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java (hunk 7 above)
Comment (CJComac) [reply caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java:159]: There is improved error handling surrounding this concern in my branch that adds support for NO_PROXY, HTTP_PROXY and HTTPS_PROXY settings.

File: caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java (hunk 8 above)
Comment (CJComac) [reply caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java:169]: We will need to revisit this in my support for NO_PROXY, HTTP_PROXY and HTTPS_PROXY branch. The url.openConnection() happens at the WebserviceClientAuditConnection level and that instance is being passed down from there.

File: caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java (hunk 9 above)
Comment (CJComac) [reply caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java:178]: This status code range logic I have written is based on Dermot's suggestions

File: caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java (hunk 10 above)
Comment (CJComac) [reply caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java:190]: We will need to revisit this in my support for NO_PROXY, HTTP_PROXY and HTTPS_PROXY branch. The url.openConnection() happens at the WebserviceClientAuditConnection level and that instance is being passed down from there.

File: pom.xml (hunk 11 above)
Comment (CJComac) [reply pom.xml:112]: TODO

File: caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java (hunk 19 above)
Comment (CJComac) [reply caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java:156]: This has been adjusted in the HTTP/HTTPS Proxy support branch

File: caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientException.java (hunk 21 above)
Comment (CJComac) [reply caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientException.java:18]: Dermot's suggestion

File: caf-audit/src/test/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditChannelTest.java (hunk 22 above)
Comment (CJComac) [reply caf-audit/src/test/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditChannelTest.java:57]: That's correct Greg. However in the HTTP/HTTPS Proxy support branch this test has been moved out into another Test class and a different test method has replaced it.

File: caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java (hunk 14 above)
Comment (CJComac) [reply caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java]: I will revisit the javadoc of this in my branch that adds support for NO_PROXY, HTTP_PROXY and HTTPS_PROXY settings as constructor parameter names have changed here

File: caf-audit-service-container/src/integration-test/java/com/hpe/caf/services/audit/api/AuditConnectionHelper.java (hunk 3 above)
Comment (CJComac) [reply caf-audit-service-container/src/integration-test/java/com/hpe/caf/services/audit/api/AuditConnectionHelper.java:46]: I will revisit the javadoc of this in my branch that adds support for NO_PROXY, HTTP_PROXY and HTTPS_PROXY settings as method parameter names have changed here

File: caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java (hunk 15 above)
Comment (CJComac) [reply caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java]: I have added a small piece of javadoc here in my latest commit but I will revisit this in my new branch, that adds support for NO_PROXY, HTTP_PROXY and HTTPS_PROXY settings, as exceptions that are thrown by this method will likely change.

File: caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java (hunk 18 above)
Comment (CJComac) [reply caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java]: Corrected in the latest commit

File: caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java (hunk 17 above)
Comment (CJComac) [reply caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java]: Corrected in the latest commit. I missed it as I went over cleanup in the code; went I first started using the JSON builder I was using it differently that how it is being used in the code now and I required it in the constructor.

File: caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java (hunk 16 above)
Comment (CJComac) [reply caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java]: Corrected in the latest commit

File: caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java (hunk 6 above)
Comment (CJComac) [reply caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java]: The idea here is that it is being reset so that the user can use the event builder instance again after calling send. Moved this into the method that returns the json string in the latest commit.

File: caf-audit-service-container/src/integration-test/java/com/hpe/caf/services/audit/api/WebserviceClientAuditIT.java (hunk 5 above)
Comment (CJComac) [reply caf-audit-service-container/src/integration-test/java/com/hpe/caf/services/audit/api/WebserviceClientAuditIT.java]: Corrected in the latest commit

File: caf-audit/src/main/java/com/hpe/caf/auditing/AuditConnectionFactory.java (hunk 4 above)
Comment (CJComac) [reply caf-audit/src/main/java/com/hpe/caf/auditing/AuditConnectionFactory.java]: Replaced with switch statement in the latest commit

File: caf-audit/src/main/java/com/hpe/caf/auditing/AuditConnectionFactory.java (hunk 12 above)
Comment (CJComac) [reply caf-audit/src/main/java/com/hpe/caf/auditing/AuditConnectionFactory.java]: Adjusted in the latest commit

File: caf-audit-service-container/src/integration-test/java/com/hpe/caf/services/audit/api/AuditConnectionHelper.java (hunk 2 above)
Comment (CJComac) [reply caf-audit-service-container/src/integration-test/java/com/hpe/caf/services/audit/api/AuditConnectionHelper.java:28]: I will revisit the javadoc of this in my branch that adds support for NO_PROXY, HTTP_PROXY and HTTPS_PROXY settings as method parameter names have changed here

File: caf-audit-service-container/pom.xml (hunk 13 above)
Comment (CJComac) [reply caf-audit-service-container/pom.xml:274]: Adjusted in the latest commit

**Comment by glucy** [comment]
Closed as umerged on purpose.  @CJComac pulled these changes into another branch and created https://github.com/CAFAudit/audit-service/pull/24
